### PR TITLE
Laura initial edits to VCS Connections page

### PR DESCRIPTION
**Pull request by laurapacilio**
### What
Edits to @matejrisek PR for Trigger Patterns Updates :) https://github.com/hashicorp/terraform-docs-common/pull/30

This PR proposes a re-write and reorganization of the entire page content. This was outside the initial scope of the PR, but I think it's important to do, so I opened this companion PR that we can merge into the original one. Rationale is that there were several parts of the page with outdated language, according to our style guide. The goal is to streamline the content and make it easier for users to read. 

One major change is that I think it needs to be easier for users to match the settings we list on this page with what they are seeing in the UI. As such, we need to make sure the section names match the UI and organize them in the order they are presented on the page.

A couple things I'm unsure of:
- I rewrote the "error handling" section, but I'm not sure what it means. Will need to discuss with engineering. 
- I'm not sure if the Working directory field is actually still there or not? 
----------

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201789378062832

### CODE REVIEW

File: website/docs/cloud-docs/run/ui.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/run/ui.mdx:30]: @matejrisek ah, I didn't realize there was already a link to the Run Triggers section that we're working on. This is perfect, I just modified it a bit so it adheres to our style guide.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:52]: @matejrisek I'm confused by this. Can we set this on this page, or do we only set it on "General Settings"?  

Regardless, the old docs had a note about how run triggering was only relevant if you hadn't specified a working directory. Should we bring back that note? If not, we can remove this content entirely. 

Here's the note for reference: 
https://www.terraform.io/cloud-docs/workspaces/settings/vcs#automatic-run-triggering

"Note: This setting only affects workspaces that specify a Terraform working directory. Without a working directory, the entire repository is considered relevant and any change will trigger a run. You can set [The working directory](https://www.terraform.io/cloud-docs/workspaces/settings#terraform-working-directory) on the **General** settings page."

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:50]: @matejrisek I don't know that we needed an entire subsection for the information about what happens if terraform can't retrieve the list of changes. So I tried putting it in a note. Thoughts?   

Also, I don't really understand what the current content means. To "All attached workspaces" --> does that mean that if TFC can't get information about a repository, it automatically just triggers runs in any/all terraform cloud workspaces that are attached to that repository? How often does this happen.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:47]: @matejrisek - Are glob patterns a well known construct? I'm wondering if we could link to external documentation about then rather than reinventing the wheel in our own documentation. But I defer to you. I don't know what the rules are/how much detail we'd need to give!

File: website/docs/cloud-docs/run/ui.mdx
Comment (matejrisek) [reply website/docs/cloud-docs/run/ui.mdx:30]:
```
If a Terraform working directory is configured, Terraform Cloud assumes that only some of the content in the repository is relevant to Terraform
```

I've missed this part here - with the introduction of `trigger-patterns` we won't insist on the dependency between working directory and triggers. Trigger prefixes on the other hand will still take working directory into consideration but we won't insist on specifying a working directory as a prerequisite to input `trigger-prefixes`.

We would probably want to remove/reword the above statement from this part of documentation.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (matejrisek) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:47]: It's been decided that we won't be linking external documentation for this part, there's [an Asana ticket](https://app.asana.com/0/1201659973165547/1202390263139034) to create documentation for glob patterns.

Glob patterns is a known construct but it's not standardized. We're only allowing for the subset of it.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (matejrisek) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:50]: We should start with `Terraform Cloud` instead of `Terraform`, since it's TFC that triggers runs :)

Aside from that it looks good to me even though I'm not completely familiar with the behavior of TFC in this case. But I'd say the semantics would have stayed the same after this change - so it's good to go.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (matejrisek) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:52]: It's not true anymore, with the introduction of `trigger-patterns` we're removing the dependency between working directory and triggers. Although if we choose `prefixes` as a type of triggers we want to use - they will still take working directory into consideration when deciding whether to trigger a run.

@omarismail and @aaabdelgany could you confirm on this? I think I've read in one of the recent conversations that we've decided to remove working directory as a prerequisite to input both `trigger-patterns` and `trigger-prefixes` in the UI.

P.S.
I think we've forgotten to update documentation on where we can set working directory - right now one can set it both through the VCS section and the General settings page.

Comment (aaabdelgany) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:52]: That is correct. A working directory is no longer necessary to add trigger patterns, but it is necessary to add trigger prefixes. We will flash a warning in the UI if the working directory is empty and they try to use trigger prefixes.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:50]:
```suggestion
-> **Note:** Terraform Cloud triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.
```

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:52]: ahhh okay! then I'm going to leave this in, and @matejrisek can draft something for it after this PR is merged.

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/settings/vcs.mdx:82]: @matejrisek added this here. Refer to my earlier comment but TLDR is that if this is really only relevant to trigger patterns and NOTHING else in TFC, maybe we should document it on this page?

File: website/docs/cloud-docs/run/ui.mdx
Comment (laurapacilio) [reply website/docs/cloud-docs/run/ui.mdx:30]: got it! I'm going to push up a change to remove!

File: website/docs/cloud-docs/workspaces/settings/vcs.mdx
Comment (matejrisek) [reply website/docs/cloud-docs/workspaces/settings/vcs.mdx:82]: It is relevant only to trigger patterns atm. We could add it here for starters. Great idea 🥇

**Review by matejrisek — APPROVED** [review]

**Pull request merged**
